### PR TITLE
fix: RBAC calls getPermissionsSummary, changes to admin's use of listRoles [WEB-517]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

The main objective / fix:

- When RBAC is on, we call useFetchMyRoles / getPermissionsSummary and update the Store with those roles [if you'd like me to call this useFetchCurrentUserRoles or something more verbose let me know]
- Loading these from the real API -> `UserAssignment` changes -> `relevantAssigned` changes

Changes to where we call `listRoles`:

- The call to useFetchKnownRoles / listRoles is moved from the main page to the UserSettings and GroupSettings pages, as it's only used by admin on these two pages to load the list of roles that Users and Groups can be assigned to

Changes to `getUserPermissions` (same as getPermissionsSummary, + a userId param) 

- We pass the `userId` to the API to get their roles instead of the current user's roles

## Test Plan

For @NicholasBlaskey who is testing this in EE:

- Can you view workspaces?
- On `/det/settings/user-management/rbac-enabled` can you see a list of permissions on each user?
- When editing that user, or a group on `/det/settings/group-management/rbac-enabled`, do you have a list of roles to assign?

## Checklist
- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.
- [x] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

File: webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx
Comment (ClaireNeveu) [inline webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx:57]: Little bit confused by this. This is the form for creating a user, right? Shouldn't we be selecting roles and not permissions?

Comment (mapmeld) [reply webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx:57]: When we view or edit the user, this populates a viewable table of their permissions (a bit controversial especially in the future where they have permissions on specific workspaces)

The roles selector gets roles from `knownRoles` which comes from the Store and is updated by `fetchKnownRoles` on the UserManagement page.  It might be movable into this component if that seems better.

Comment (ClaireNeveu) [reply webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx:57]: If it's just for display then we really should be deriving those permissions in the render function but that's not important for this PR.

File: webui/react/src/services/api.ts
Comment (ClaireNeveu) [inline webui/react/src/services/api.ts:86]: Was this type just wrong before?

Comment (ClaireNeveu) [reply webui/react/src/services/api.ts:86]: Okay I see we were throwing out information in the API response parsing step and now we're not.

File: webui/react/src/services/apiConfig.ts
Comment (ClaireNeveu) [inline webui/react/src/services/apiConfig.ts:206]: This name doesn't make it clear what this does. Can we add a JSDoc explaining that this fetches the permissions summary for the currently logged-in user?

File: webui/react/src/services/decoder.ts
Comment (ClaireNeveu) [inline webui/react/src/services/decoder.ts:52]: We shouldn't use 0 as a special value. If this field is optional the type should be nullable.

File: webui/react/src/services/decoder.ts
Comment (ClaireNeveu) [inline webui/react/src/services/decoder.ts:53]: Just curious here why you do `new Array<number>()` instead of `[]`?

File: webui/react/src/services/decoder.ts
Comment (mapmeld) [reply webui/react/src/services/decoder.ts]: Backend did not do the annotation to require fields on their custom proto types for RBAC, only the request/response objects. Saw this before with Role. Just pushed a fix to this one.

File: webui/react/src/services/decoder.ts
Comment (mapmeld) [reply webui/react/src/services/decoder.ts:53]: I thought that TypeScript would care about the type of the empty array, but (in this context at least?) it doesn't need a type hint. Will do `[]` here

File: webui/react/src/services/apiConfig.ts
Comment (mapmeld) [reply webui/react/src/services/apiConfig.ts:215]: Added JSDoc here and to the similar `getUserPermissions` in the latest commit; hope that helps.

**Review by ClaireNeveu — APPROVED** [review]
This looks good on my end but please get a review from @NicholasBlaskey before merging.

**Review by NicholasBlaskey — APPROVED** [review]

**Pull request merged**
